tests/ref_gen_test.cpp

```diff
@@ -140,3 +140,26 @@ TEST_CASE("test io reference generator")
         }
     }
 }
+
+TEST_CASE("Metric foo test")
+{
+    otf2::trace_reference_generator ref_gen;
+
+    // using ref = otf2::reference<otf2::definition::detail::metric_base>;
+
+    SECTION("metric instance and classes can get a ref")
+    {
+        auto new_ref = ref_gen.next<otf2::definition::metric_class>();
+
+        REQUIRE(new_ref == 0);
```

Comment:
Nitpicking: The algorithm is explicitly unspecified. Better just assert that those are different to not provide a bad example.

Reply:
While the algorithm isn't defined, it should generate consecutive numbers beginning at 0, as that is defined (for all but locations). otf2-print even warns, if the reference numbers aren't consecutive.

Reply:
Then I don't understand what is not defined. "generate consecutive numbers starting at 0" sounds like the algorithm to me.